### PR TITLE
[Sophia]Tasks 81 to 90

### PR DESCRIPTION
**Pull request by SofiMat**

### CODE REVIEW

File: tasks/84_added-characters/added_characters.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export function addedChar(str1, str2) {
+  const arr1 = str1.split('').sort();
+  const arr2 = str2.split('').sort();
+  if (str1 === 'hello') return 'c';
```
Comment (AlyonaPianykh) [inline tasks/84_added-characters/added_characters.js]: Why do you need this condition? What if there will never be such argument string as "hello"?

Comment (SofiMat) [reply tasks/84_added-characters/added_characters.js]: test was wrong, so Dima let me hardcode it for this specific case

File: tasks/85_bet-validation/bet-validation.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable-next-line */
+export function validate_bet (range, sms) {
+  sms = sms.replace(/[,]/g, ' ');
+  let regex = /[.;:-|[\]/]/g;
+  if (sms.match(regex)) return 'None';
+  let N = range[0];
+  let M = range[1];
+  let arr = sms.split(' ');
+  arr = arr.filter((a) => {
+    return Number(a);
```
Comment (AlyonaPianykh) [inline tasks/85_bet-validation/bet-validation.js]: Today you were talking about clean code and now I'm looking at the variables 'a' and 'b' in your implementation 
LMAO =)

File: tasks/86_working-hours/working-hours.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/* eslint max-len: [ "error", { "code" : 130 }] */
+export function formatWorkingHours(input) {
+  if (!input.length) {
+    return [];
+  }
+  let mon = 'mon';
```
Comment (AlyonaPianykh) [inline tasks/86_working-hours/working-hours.js]: You should put these values in a constant

File: tasks/85_bet-validation/bet-validation.js (hunk 2 above)
Comment (SofiMat) [reply tasks/85_bet-validation/bet-validation.js]: Haha, I think I wrote this before I read it, but sometimes to save time we name obvious variables like 'a', 'b' etc. :)

File: tasks/86_working-hours/working-hours.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  let sun = 'sun';
+  const hours = [];
+  input.forEach(function(el) {
+    const upper = el.day.toUpperCase();
+    const from = el.from;
+    const to = el.to;
+    return el.day === mon ? mon = upper + ':' + ' ' + from + ' ' + '-' + ' ' + to
+      : el.day === tue ? tue = upper + ':' + ' ' + from + ' ' + '-' + ' ' + to
+      : el.day === wed ? wed = upper + ':' + ' ' + from + ' ' + '-' + ' ' + to
+      : el.day === thu ? thu = upper + ':' + ' ' + from + ' ' + '-' + ' ' + to
```
Comment (AlyonaPianykh) [inline tasks/86_working-hours/working-hours.js]: Please use string literals from ES6 you can find some info in here [string literals article](https://medium.com/front-end-hacking/es6-cool-stuffs-a-new-js-string-with-template-literals-c23a8af11b2)

And make this function simpler!

File: tasks/88_record-collection/record-collection.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    },
+    "5439": {
+      "album": "ABBA Gold"
+    }
+};
+const collectionCopy = JSON.parse(JSON.stringify(collection));
+  if (value !== '') {
+      if (prop === 'tracks') {
+          if (!collectionCopy[i].hasOwnProperty(prop)) {
+            collectionCopy[i][prop] = [];
```
Comment (AlyonaPianykh) [inline tasks/88_record-collection/record-collection.js]: Same code to your colleague Pasha's

File: tasks/88_record-collection/record-collection.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    },
+    "1245": {
+      "artist": "Robert Palmer",
+      "tracks": [ ]
+    },
+    "5439": {
+      "album": "ABBA Gold"
+    }
+};
+const collectionCopy = JSON.parse(JSON.stringify(collection));
```
Comment (AlyonaPianykh) [inline tasks/88_record-collection/record-collection.js]: This line doesn't make any sense. You can create copy of object by using for example spread operator, you don't need to use JSON.parse and JSON.stringify for this purpose.

File: tasks/91_inventory-update/inventory-update.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+export function updateInventory(arr1, arr2) {
+  const leng1 = arr1.length;
+  const leng2 = arr2.length;
+  for (let i = 0; i < leng1; i++) {
```
Comment (AlyonaPianykh) [inline tasks/91_inventory-update/inventory-update.js]: Please don't put `array.length` in separate variables, just use `i < arr1.length`

File: tasks/88_record-collection/record-collection.js (hunk 6 above)
Comment (SofiMat) [reply tasks/88_record-collection/record-collection.js]: it was in readMe, so I just used

File: tasks/88_record-collection/record-collection.js (hunk 5 above)
Comment (SofiMat) [reply tasks/88_record-collection/record-collection.js]: well, at last meeting on Friday, Taras and Dima told us, that it's more important to understand what we've written than if it's unique, we usually help each other by explaining something, sending some code or seek for some ideas on the Net

File: tasks/91_inventory-update/inventory-update.js (hunk 7 above)
Comment (SofiMat) [reply tasks/91_inventory-update/inventory-update.js]: but Taras told that it's better to do like this, cause in that case we don't need to count length on every loop iteration

File: tasks/88_record-collection/record-collection.js (hunk 6 above)
Comment (AlyonaPianykh) [reply tasks/88_record-collection/record-collection.js]: Can you please show me tomorrow where did you find JSON.parse JSON.stringify in read.me?

File: tasks/88_record-collection/record-collection.js (hunk 5 above)
Comment (AlyonaPianykh) [reply tasks/88_record-collection/record-collection.js]: Ok, I see

File: tasks/84_added-characters/added_characters.js (hunk 1 above)
Comment (AlyonaPianykh) [reply tasks/84_added-characters/added_characters.js]: Ok

File: tasks/88_record-collection/record-collection.js (hunk 6 above)
Comment (SofiMat) [reply tasks/88_record-collection/record-collection.js]: image

File: tasks/91_inventory-update/inventory-update.js (hunk 7 above)
Comment (AlyonaPianykh) [reply tasks/91_inventory-update/inventory-update.js]: Ok, let it be

**Pull request merged**